lib/elixir_sso_example_web/controllers/page_controller.ex

```diff
@@ -1,23 +1,39 @@
 defmodule ElixirSSOExampleWeb.PageController do
   use ElixirSSOExampleWeb, :controller
 
-  def index(conn, %{"code" => code}) do
+  def index(conn, _params) do
+    isLoggedIn = get_session(conn, :isLoggedIn)
+    profile = get_session(conn, :profile)
+    isLoggedIn && render(conn, "login_successful.html", profile: profile)
+
+    render(conn, "index.html")
+  end
+
+  def login(conn, _params) do
+    {:ok, redirect_url} = WorkOS.SSO.get_authorization_url(%{
+
+      connection: Application.get_env(:elixir_sso_example, :connection),
+      redirect_uri: Application.get_env(:elixir_sso_example, :redirect_uri)
+    })
+    redirect(conn, external: redirect_url)
+  end
+
+  def callback(conn, %{"code" => code}) do
     case WorkOS.SSO.get_profile(code) do
       {:ok, %{"profile" => profile}} ->
-        render(conn, "index.html", profile: profile)
+        conn = put_session(conn, :isLoggedIn, true)
```

Comment:
I think we could use pipe here instead of passing the function value directly to variables, it follows more the data transformation style of Elixir

```ex
conn
|> put_session(:isLoggedIn, true)
|> put_session(:profile, profile)
|> redirect(to: "/")
```
```
```